src/box/session.h

```diff
@@ -375,6 +375,69 @@ generic_session_fd(struct session *session);
 int64_t
 generic_session_sync(struct session *session);
 
+/** \cond public */
+
+/**
+ * Get the unique identifier (ID) for the current session.
+ *
+ * @retval session ID
+ */
+API_EXPORT int
+box_session_id(void);
+
+/**
+ * Check if session exists.
+ *
+ * @param sid session ID
+ * @retval true if sessions exists, false otherwise
+ */
+API_EXPORT bool
+box_session_exists(uint64_t sid);
+
+/**
+ * Get host address and port of the session peer (if it exists).
+ *
+ * @param sid session ID
+ * @retval peer's host and port on success, NULL on error.
+ */
+API_EXPORT const char *
+box_session_peer(uint64_t sid, char *dst, size_t size);
```

Comment:
Hm. Did you think about using struct sockaddr as an out parameter? Maybe the user does not want a string. But if he wants, he can just do his own conversion. This would help us to get rid of the buffer copying problem. Just make the function work like `getpeername()`. So it takes a session ID, `struct sockaddr *addr`, and `uint32_t *addrlen`. It returns `int`. If 0, it was a success, and the address is filled with the peer address. Addrlen is an in-out parameter. After the call it contains the number of bytes written to addr. Just see `man getpeername` for details.

Note, I intentionally didn't use `socklen_t`, because it would force us to include socket headers into module.h. But with the proposed way you can just announce the sockaddr struct in module.h like this: `struct sockaddr;`.

Reply:
I moved to implementation of `box_session_peer` from old PR (the same, that you are proposing now).

Reply:
Hm. I don't remember you using `struct sockaddr` in any of your old PRs. Could you please give a link?